sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -844,39 +885,20 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
       KSId = getNextKernelSetId();
 
     auto &Imgs = m_DeviceImages[KSId];
-    if (!Imgs)
-      Imgs.reset(new std::vector<DeviceImage *>({Img}));
-    else
-      Imgs->push_back(Img);
+    if (!Imgs) {
+      Imgs.reset(new std::vector<RTDeviceBinaryImageUPtr>());
+      Imgs->push_back(std::move(Img));
+    } else
+      Imgs->push_back(std::move(Img));
```

Comment:
I suggest removing the else branch and moving the `push_back` line outside of `if`.

Reply:
OK